src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -304,6 +312,8 @@ void GIVEN_managedCAConfiguration_WHEN_updatedToCustomCAConfiguration_THEN_serve
 
         URI privateKeyUri = new URI("file:///private.key");
         URI certificateUri = new URI("file:///certificate.pem");
+        when(securityServiceMock.getKeyPair(privateKeyUri, certificateUri)).thenReturn(intermediateKeyPair);
+        doReturn(chain).when(certificateStoreSpy).loadCaCertificateChain(privateKeyUri, certificateUri);
 
         AtomicReference<CertificateUpdateEvent> eventRef = new AtomicReference<>();
         Pair<CompletableFuture<Void>, Consumer<CertificateUpdateEvent>> asyncCall =
```

Comment:
I realize this is a smaller refactor, but I'll just make a note of this.

This test function is really confusing IMO, and I'm really failing to see how it makes anything easier. It just seems to add layers of indirection. Also, I don't think it's correct to assert on how many times the consumer was "accepted"

I'd vote for removing it from our tests and make whatever this is doing under the hood explicit. Thoughts?